src/test/java/plugins/MSBuildPluginTest.java

```diff
@@ -0,0 +1,133 @@
+package plugins;
+
+import static org.junit.Assert.assertThat;
+
+import java.io.IOException;
+
+import org.apache.commons.lang3.SystemUtils;
+import org.hamcrest.Matchers;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildInstallation;
+import org.jenkinsci.test.acceptance.msbuild.MSBuildStep;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.internal.AssumptionViolatedException;
+
+/**
+ * This test is designed to be run on a Windows machine with MSBuild installed
+ * 
+ * In order for the test to know where the MSBuild installation dir is, an
+ * environment variable must be set. i.e:
+ * 
+ *      MSBUILD_EXE = C:\Windows\Microsoft.NET\Framework\v2.0.50727\MSBuild.exe
+ * 
+ * Tests will be skipped if:
+ *      - The environment variable is not set 
+ *      - The test is not running on a Windows machine
+ *      - MSBUild.exe is not present in the specified location
+ */
+@WithPlugins("msbuild")
+public class MSBuildPluginTest extends AbstractJUnitTest {
+
+    private static final String MSBUILD_EXE_ENV = "MSBUILD_EXE";
+    private static final String MSBUILD_NAME = "MSBuildInstallation";
+
+    /**
+     * Builds a FreeStyle job with a MSBuild step with the configuration passed as parameter
+     * 
+     * @param workspacePath The workspace where the project is located
+     * @param buildFile The build file (.proj or .sln)
+     * @param cmdArguments the command line arguments to execute. Can be null.
+     * @return a job
+     * @throws IllegalArgumentException if workspacePath or buildFile are null
+     */
+    private FreeStyleJob msBuildJob(String workspacePath, String buildFile, String cmdArguments) {
+        if (workspacePath != null && buildFile != null) {
+            FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class);
+            job.copyDir(resource(workspacePath));
+            MSBuildStep msBuildStep = job.addBuildStep(MSBuildStep.class);
+            msBuildStep.setMSBuildName(MSBUILD_NAME).setMSBuildFile(buildFile);
+            if (cmdArguments != null && !cmdArguments.isEmpty()) {
+                msBuildStep.setCmdLineArgs(cmdArguments);
+            }
+            job.save();
+            return job;
+        } else {
+            throw new IllegalArgumentException("Workspace and buildFile must be different from null.");
+        }
+    }
+    
+    @Before
+    public void setUp() {
+         //Check environment configuration
+         if (System.getenv(MSBUILD_EXE_ENV) == null) {
```

Comment:
@olivergondza I am trying to use the annotation you suggested but I am seeing something strange.. From [this](https://github.com/jenkinsci/acceptance-test-harness/blob/b27c150f35a386dfced4a5960127bf62f9c34363/src/main/java/org/jenkinsci/test/acceptance/junit/TestActivation.java#L82), it seems like the environment variable that needs to be present must look like `{className.envVarName}`. I see two problems here:

- Unix does not allow dots in environment variables. See [here](http://unix.stackexchange.com/questions/93532/exporting-a-variable-with-dot-in-it#answer-93533)
- How can tests like [this](https://github.com/jenkinsci/acceptance-test-harness/blob/b27c150f35a386dfced4a5960127bf62f9c34363/src/test/java/core/JdkTest.java) be working. First, we have the issue with the dot. Second, look at line [15](https://github.com/jenkinsci/acceptance-test-harness/blob/b27c150f35a386dfced4a5960127bf62f9c34363/src/test/java/core/JdkTest.java#L15), the variable that tries to get does not include the `className.`before the environment variable name. So, for this test, there has to be two environment variables set. One with the class name (to pass the activation) and one without the dot (so that the value can be retrieved). And in Unix, the one with the dot cannot be set.

Any thoughts? Am I missing something?